p2p/auth.py

```diff
@@ -78,6 +79,11 @@
         token=token,
         timeout=REPLY_TIMEOUT)
 
+    if reader.at_eof():
+        # This is what happens when Parity nodes have blacklisted us
+        # (https://github.com/ethereum/py-evm/issues/901).
```

Comment:
I'm not a fan of linking out to github from a comment usually, but I may be alone on that. My logic is that putting the link can be used as a crutch to avoid fully describing what's going on. But most people won't follow the link, so they end up without as clear a picture of the situation.

Reply:
Right, but I think the code here is straightforward enough to be understood even without the comment. The only reason I added the comment is so that, if necessary, one can understand why it had to be written in the first place